app/models/post.rb

```diff
@@ -0,0 +1,16 @@
+class Post < ApplicationRecord
+  after_create :update_posts_counter
+  after_destroy :update_posts_counter
+
+  belongs_to :author, class_name: 'User', foreign_key: 'author_id', inverse_of: 'posts', counter_cache: :posts_counter
+  has_many :comments
+  has_many :likes
+
+  def most_recent_comments(number = 5)
+    comments.order(created_at: :desc).limit(number)
+  end
+
+  def update_posts_counter
+    author.update(posts_counter: author.posts.count)
+  end
+end
```

Comment:
### Optional

- [ ] 🎉 Great job on the PR! 👏 Your code looks good, and the functionality is in place. I noticed an alternative approach for updating posts_counter that might improve efficiency.

- [ ] Consider using` author.increment!(:posts_counter)` instead of fetching the count and then updating it. This method directly increments the counter in the database, reducing the need for an extra query. It's a more efficient and idiomatic way in ActiveRecord.

#### example

```
def update_posts_counter
  author.increment!(:posts_counter)
end
```

- [ ]  Feel free to incorporate this change if you find it suitable. Keep up the good work! 🚀 If you have any questions or need further clarification, feel free to ask. 🤓

Reply:
Thanks for your suggestion @frank1738, I would like to ask something.
What about when you delete a post? Do I have to do another custom method so the posts_counter will continue updated?

Reply:
> Thanks for your suggestion @frank1738, I would like to ask something. What about when you delete a post? Do I have to do another custom method so the posts_counter will continue updated?

@luigirazum . Yes, you can create another custom method that uses `author.decrement!(:posts_counter)`

Reply:
All right @frank1738, thanks again for your suggestions. I'll apply them for the next code review.